firmware/src/core/utils/HTTPClientWrapper.cpp

```diff
@@ -0,0 +1,167 @@
+#include "HTTPClientWrapper.h"
+
+HTTPClientWrapper* HTTPClientWrapper::instance = nullptr;
+SemaphoreHandle_t HTTPClientWrapper::requestSemaphore = nullptr;
+QueueHandle_t HTTPClientWrapper::requestQueue = nullptr;
+QueueHandle_t HTTPClientWrapper::responseQueue = nullptr;
+
+// Initialize static debug counters
+volatile uint32_t HTTPClientWrapper::activeRequests = 0;
+volatile uint32_t HTTPClientWrapper::maxConcurrentRequests = 0;
+
+HTTPClientWrapper::HTTPClientWrapper() {
+    if (!requestSemaphore) {
```

Comment:
Additionally, if these are just members of the `HTTPClientWrapper` there is no need to do these checks both here and in the destructor. (That need already wasn't there btw, since it is a singleton)

Reply:
yeah. destructor never gets used since it a singleton and lives with the app

Reply:
destructor removed